network-api/networkapi/wagtailpages/pagemodels/buyersguide/products.py

```diff
@@ -1026,7 +1029,24 @@ def localized_related_products(self):
         )
         related_products = localize_queryset(related_products, preserve_order=True)
         return related_products.specific()
-
+    
+    @property
+    def preview_related_products(self) -> list:
+        """
+        Fetches related product updates for CMS page previews.
+        """
+        related_products = orderables.get_related_items(
+            self.related_product_pages.all(),
+            "related_product",
+        )
+        # FIXME: This implementation does return the localized version of each product.
+        #        But, it is inefficient. It would be better to pull all products
+        #        for the correct locale at once. This would require the above returns
+        #        a queryset of the products (rather than a list) and that we have an
+        #        efficient way of pulling all items for a given locale.
+        #        See: https://github.com/MozillaFoundation/foundation.mozilla.org/issues/9509
```

Comment:
We can change this comment for some more appropriate for this method to the tune of "we know this is inefficient but that's on purpose as we need to make sure that the preview renders the objects in memory"

Reply:
Done!